core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -370,6 +383,20 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
 
       purged
     }
+
+    def forceComplete(): Int = {
```

Comment:
One way to do this is to first mark the coordinator as not available, then simply call Purgatory.checkAndComplete(). Those pending produce requests are guaranteed to complete since those pending partitions will error out after checking the coordinator state in tryComplete(). Then, we don't need this forceComplete() call. We probably need to call Purgatory.checkAndComplete() while holding the coordinator lock so that we can guarantee that the coordinator state doesn't change when the purgatory does the check.